### PR TITLE
Offline feed efficacy vd tests

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#28210|

## Description 

- Added a reduced content file from current snapshot (vd_1.0.0_vd_4.8.0_1317226_1740423267.json)
- Generate database in the same workflow 
- Restored some of the old tests
- Added Readme.md files on each tests with a description

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/008/Readme.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+- Vulnerability detection validation, vulnerability remotion after upgrade and entry deletion after `integrity_clear` event.
+
+## Platforms
+
+- Microsoft Windows 10 19045.4043
+- Microsoft Windows 10 19045.4046
+
+## Checks performed
+
+- CVE-2024-23224 does affect `MacOS` OS version `14.0`.
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/008/Readme.md:12]: We need to update the `macOS` references in this Windows test.

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/010/input_001.json
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/010/input_001.json:1]: I consider the other Readme.md files are valuable.
I'd add one in this test.

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/011/input_001.json
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/011/input_001.json:1]: I consider the other Readme.md files are valuable.
I'd add one in this test.

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/011/expected_003.out
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/011/expected_003.out:1]: +1 Good feature testing with this test.

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/010/expected_002.out
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/010/expected_002.out:1]: +1 Good feature testing with this test.

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp [hunk 2]
```diff
@@ -48,60 +48,64 @@ class TCVESolvedInventorySync final
      */
     std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
     {
-        // We look for the CVEs in the agent's vulnerabilities inventory
-        const auto& column = AFFECTED_COMPONENT_COLUMNS.at(AffectedComponentType::Package);
-
-        for (const auto& [key, value] : TInventorySync<TScanContext>::m_inventoryDatabase.seek(data->agentId(), column))
+        // We look for the CVEs in the agent's vulnerabilities inventory for OS and Packages
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp:51]: We should discuss with the team about this use case.
I understand that it wasn't in the requirements of
https://github.com/wazuh/wazuh/issues/23066

A hotfix insertion will eventually be reflected for the OS CVEs in the next OS scan. But it isn't a bad idea to react faster.

**Review by pereyra-m — COMMENTED** [review]
Gj!
Just come comments

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp (hunk 2 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp:51]: The requirements were too open, I had to test as much as possible the behavior of the scanner with fixed content. 

About this 

> A hotfix insertion will eventually be reflected for the OS CVEs in the next OS scan. But it isn't a bad idea to react faster.

But that new OS scan won't happen after a restart? we are ignoring OS delta events for obvious reason. 

I'll let this as it is now, until we define whether or not I should remove this. 

CC: @Dwordcito

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/008/Readme.md (hunk 1 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/qa/test_data/008/Readme.md:12]: Oops!

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/011/input_001.json
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/qa/test_data/011/input_001.json:1]: Yeap, I focused too much in the behavior and bug fixes and overlooked the readme files at the end. I'll revise them. Thanks!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp:51]: > The requirements were too open, I had to test as much as possible the behavior of the scanner with fixed content.

You've selected a wide range of tests.
I meant that when we developed that feature, we focused on the packages affected by the hotfixes only.

> But that new OS scan won't happen after a restart? we are ignoring OS delta events for obvious reason.

The OS should be re-scanned in every syscollector scan.

> I'll let this as it is now, until we define whether or not I should remove this.

Sure, we wait for definitions.

Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/cveSolvedInventorySync.hpp:51]: >  I meant that when we developed that feature, we focused on the packages affected by the hotfixes only.

Did we really focus on that? or did we forget to update it there after we changed it here https://github.com/wazuh/wazuh/pull/23088?

**Review by GabrielEValenzuela — APPROVED** [review]
I believe we have covered most cases related to the efficacy tests, including common operating systems like Ubuntu and Windows, as well as edge cases such as Windos hotfixes and Arch. In the near future, we could consider adding RockyLinux and Windows Server to the test too.

Regarding the Python script, the process wait mechanism is now clearer and easier to understand.

As for the change in the scanner, I think it is valid—hotfixes can affect both individual packages and the operating system itself. I believe this was previously discussed with @Dwordcito 

All tests have passed, and everything is working as expected. Great work, Migue!

**Review by pereyra-m — APPROVED** [review]
